app/src/main/java/com/smelnykov/homework_1/MyProfileActivity/MyProfileActivity.kt

```diff
@@ -32,4 +44,34 @@ class MyProfileActivity : AppCompatActivity() {
         val name = binding.appCompatTextName
         name.text = parser.parseNameSurname(email)
     }
+
+    /**
+     * Method sets up the click listeners for log out button.
+     */
+    private fun setClickListeners() {
+
+        val buttonLogOut = binding.appCompatButtonLogOut
+
+        buttonLogOut.setOnClickListener {
+            clearData()
+            logOutRedirection()
+        }
+    }
+
+    /**
+     * Method clears users data in sharedPreference.
+     * */
+    private fun clearData() {
+        val editor = sharedPreference.edit()
+        editor.clear()
+        editor.apply()
+    }
+
+    /**
+     * Method for changing activity to Authorization activity.
+     * */
+    private fun logOutRedirection() {
+        val intent = Intent(this@MyProfileActivity, AuthActivity::class.java)
+        startActivity(intent)
```

Comment:
it would probably return here on backpress without clearing current activity, please verify

Reply:
Yes, you are right. Thank you! Made changes. Commit 9e0bcee7